examples/src/fungible-token.js

```diff
@@ -1,87 +1,136 @@
-import {
-  NearBindgen,
-  call,
-  view,
-  initialize,
-  near,
-  LookupMap,
-  assert,
-} from "near-sdk-js";
+import { NearBindgen, call, view, initialize, near, LookupMap, assert } from "near-sdk-js";
 
 @NearBindgen({ initRequired: true })
 export class FungibleToken {
-  constructor() {
-    this.accounts = new LookupMap("a");
-    this.totalSupply = 0n;
-  }
+  accounts = new LookupMap("a");
+  accountRegistrants = new LookupMap("r");
+  accountDeposits = new LookupMap("c");
+  totalSupply = "0";
 
   @initialize({})
-  init({ prefix, totalSupply }) {
-    this.accounts = new LookupMap(prefix);
-    this.totalSupply = BigInt(totalSupply);
-    this.accounts.set(near.signerAccountId(), this.totalSupply);
-    // In a real world Fungible Token contract, storage management is required to denfense drain-storage attack
+  init({ owner_id, total_supply }) {
+    assert(BigInt(total_supply) > BigInt(0), "Total supply should be a positive number");
```

Comment:
This a good move! Let's check owner_id as well (we have a function for that in utils).

Reply:
Added! 🙌